src/wallet.js

```diff
@@ -0,0 +1,273 @@
+  }
+
+  /**
+   * Initialize a blockstack wallet
+   * @param {string} keychain - the Base58 string used to initialize
+   *  the root node of the hierarchical wallet
+   * @return {BlockstackWallet} the constructed wallet
+   */
+  static fromBase58(keychain: string): BlockstackWallet {
+    return new BlockstackWallet(HDNode.fromBase58(keychain))
+  }
+
+  getIdentityPrivateKeychain(): HDNode {
+    return this.rootNode
+      .deriveHardened(IDENTITY_KEYCHAIN)
+      .deriveHardened(BLOCKSTACK_ON_BITCOIN)
+  }
+
+  getBitcoinPrivateKeychain(): HDNode {
+    return this.rootNode
+      .deriveHardened(BITCOIN_BIP_44_PURPOSE)
+      .deriveHardened(BITCOIN_COIN_TYPE)
+      .deriveHardened(BITCOIN_ACCOUNT_INDEX)
+  }
+
+  getBitcoinNode(addressIndex: number, chainType: string = EXTERNAL_ADDRESS): HDNode {
+    return BlockstackWallet.getNodeFromBitcoinKeychain(
+      this.getBitcoinPrivateKeychain().toBase58(),
+      addressIndex, chainType)
+  }
+
+  getIdentityAddressNode(identityIndex: number): HDNode {
+    const identityPrivateKeychain = this.getIdentityPrivateKeychain()
+    return identityPrivateKeychain.deriveHardened(identityIndex)
+  }
+
+  static getAppsNode(identityNode: HDNode): HDNode {
+    return identityNode.deriveHardened(APPS_NODE_INDEX)
+  }
+
+  /**
+   * Get a salt for use with creating application specific addresses
+   * @return {String} the salt
+   */
+  getIdentitySalt(): string {
+    const identityPrivateKeychain = this.getIdentityPrivateKeychain()
+    const publicKeyHex = getNodePublicKey(identityPrivateKeychain)
+    return crypto.createHash('sha256').update(publicKeyHex).digest('hex')
+  }
+
+  /**
+   * Get a bitcoin receive address at a given index
+   * @param {number} addressIndex - the index of the address
+   * @return {String} address
+   */
+  getBitcoinAddress(addressIndex: number): string {
+    return this.getBitcoinNode(addressIndex).getAddress()
+  }
+
+  /**
+   * Get the private key hex-string for a given bitcoin receive address
+   * @param {number} addressIndex - the index of the address
+   * @return {String} the hex-string. this will be either 64
+   * characters long to denote an uncompressed bitcoin address, or 66
+   * characters long for a compressed bitcoin address.
+   */
+  getBitcoinPrivateKey(addressIndex: number): string {
+    return getNodePrivateKey(this.getBitcoinNode(addressIndex))
+  }
+
+  /**
+   * Get the root node for the bitcoin public keychain
+   * @return {String} base58-encoding of the public node
+   */
+  getBitcoinPublicKeychain(): string {
+    return this.getBitcoinPrivateKeychain().neutered().toBase58()
+  }
+
+  /**
+   * Get the root node for the identity public keychain
+   * @return {String} base58-encoding of the public node
+   */
+  getIdentityPublicKeychain(): string {
+    return this.getIdentityPrivateKeychain().neutered().toBase58()
+  }
+
+  static getNodeFromBitcoinKeychain(keychainBase58: string, addressIndex: number,
```

Comment:
This should be an instance method that just takes index and chainType, and uses `this.rootNode.derive`